### PR TITLE
Actualize Hanami::Environment comments about .env file

### PR DESCRIPTION
**Pull request by linchus**
Update documentation in Hanami::Environment to actualize dotenv file inheritance information.

Based on discussion  in #651 and #772

### CODE REVIEW

File: lib/hanami/environment.rb [hunk 1]
```diff
@@ -140,12 +140,14 @@ class Environment
...
     # such as `RACK_ENV` and `HANAMI_ENV`.
     #
-    # It also evaluates configuration from `.env` and `.env.<environment>`
+    # It evaluates configuration ONLY from `.env.<environment>` file
     # located under the config directory. All the settings in those files will
     # be exported as `ENV` variables.
     #
-    # The format of those `.env` files is compatible with `dotenv` and `foreman`
-    # gems.
+    # Master `.env` file is ignored for security reason.
```
Comment (kaikuchn) [inline lib/hanami/environment.rb]: Will people lacking the context of the discussions that lead to this PR understand this?

Comment (linchus) [inline lib/hanami/environment.rb]: I'm not sure that it is the right place to put full explanation why it is bad.
I'd say it's enough to understand code behaviour.

What text do you suggest to add here to improve understanding?

Comment (kaikuchn) [inline lib/hanami/environment.rb]: What about something along the lines of this
> Master `.env` file is ignored to suggest clear separation of environment configurations and discourage putting sensitive information into source control.

Comment (jodosha) [inline lib/hanami/environment.rb]: I'd mention also that `.env` files are only for development and not for production.

**Comment by mereghost** [comment]
@TiteiKo can you check this, and if okay, merge it?

File: lib/hanami/environment.rb [hunk 2]
```diff
@@ -140,12 +140,16 @@ class Environment
...
     # be exported as `ENV` variables.
     #
-    # The format of those `.env` files is compatible with `dotenv` and `foreman`
-    # gems.
+    # Master .env file is ignored to suggest clear separation of environment
+    # configurations and discourage putting sensitive information into source
+    # control.
+    #
+    # The format of those `.env.<environment>` files is compatible with `dotenv`
+    # and `foreman` gems.
```
Comment (TiteiKo) [inline lib/hanami/environment.rb]: dotenv and foreman, yes, but it's also compatible with systemd EnvironmentFile format and probably a lot of others, I'd add that it's a universal unix environment variable declaration format

**Comment by davydovanton** [comment]
@TiteiKo hey, any progress here? :)

**Comment by TiteiKo** [comment]
@davydovanton well I'm waiting on anybody to agree or disagree on my comment. I'm not ok merging it while it's not addressed :)

@linchus WDYT?

**Comment by linchus** [comment]
@TiteiKo What do you think about my update? Is it ok?

**Comment by b264** [comment]
Maybe could add some more info about jodosha's comment  `I'd mention also that .env files are only for development and not for production.`  but specifically also Why.  Could we add a link to why this is discouraged for further reading if they want. (even if on another site)

**Comment by TiteiKo** [comment]
LGTM :+1: 

@b264 To me, this comment should be in a guide and not in the source code: If you're not using Heroku or other similar service, chances are that your deployment strategy will generate an .env file, to be read by your service (for example using the `EnvironmentFile` directive in systemd). 

Because in truth, `.env` files were not invented by `Dotenv`, but are a common thing in UNIX-based systems, so in a way, yes, `.env` files are for production. **With a huge but**: production environment variables should **never** be committed in the same repository as the source code. 

But it **should** be in a guide :)

**Comment by b264** [comment]
Yes, just a lot of folks get confused when they see lots of "Don't do thing X" but never any "Do thing Y instead" or at a minimum a link to somewhere else that explains it in more details

**Comment by b264** [comment]
@TiteiKo If it's to be in a guide, they may not be in the guide when they see this.  But if this were to _link_ to a guide that explained it, that's kind-of what I mean :-)

**Pull request merged**